### ISSUE TITLE
Use DayNight theme

### ISSUE THREAD

**Issue by svenoaks** [geecko86/QuickLyric#198]
Instead of annoying user to say what time day and night is, use AppCompat DayNight theme. Change 'Theme' setting to read 'Colors', and 'Night Mode' to 'Theme' with Light/Dark/Auto options, with Auto as the default.

**Comment by geecko86** [comment]
Agreed. Let's do it like you described, but with Light/Dark/Auto/Custom options.

**Comment by svenoaks** [comment]
Currently it seems night mode (when active) has the same colors for all themes, is that the behavior you want to keep?

**Comment by geecko86** [comment]
Yeah but if you think there's room for improvement, we can discuss that too. What sort of ideas did you have?

**Comment by svenoaks** [comment]
I mean keep the action and status bar and accent colors of the theme color picked, instead of going to all black.

**Comment by geecko86** [comment]
I think it wouldn't look good. Do you disagree?

**Comment by svenoaks** [comment]
Not really, some of them will look bad. I'll keep as is for now.

**Comment by svenoaks** [comment]
My idea is to keep the switch for enabling the night mode, then have a list preference underneath it to select Auto or Custom times, with Auto being the default. Perhaps we could migrate those users who have selected times to the Custom setting. 

I would suggest enabling Auto night mode for unpaid users who see ads, because it will need the location access for determining the times. This will increase eCPM of your impressions.

**Comment by geecko86** [comment]
I like it, that's pretty smart. How about using [\<uses-permission-sdk-23\>](https://developer.android.com/guide/topics/manifest/uses-permission-sdk-23-element.html) so that the permission doesn't show up in the list when people with API < 23 want to install the app? We might change it at some point, but not at first, as it could affect the conversion rate.

According to the doc:

> The calculation used to determine whether it is night or not makes use of the location APIs (if this app has the necessary permissions). This allows us to generate accurate sunrise and sunset times. If this app does not have permission to access the location APIs then we use hardcoded times which will be less accurate.

Less than 20% of our users run API < 23. They would have hardcoded times, I think I'm ok with that.

> Perhaps we could migrate those users who have selected times to the Custom setting.

Could you clarify what you mean by that?

**Comment by svenoaks** [comment]
> Perhaps we could migrate those users who have selected times to the Custom setting.
> 
> Could you clarify what you mean by that?

I mean when users install the update, and they had already set times with night mode, they will using the Custom option now with those times. This might be what to do, since if change them to auto mode, they would have to approve location access out of the blue.

**Comment by geecko86** [comment]
Oh Sure. That sounds like the right thing to do.

**Comment by geecko86** [comment]
It's usually a good idea to give me the new strings as soon as possible, as the localization can sometimes take a while.

**Comment by svenoaks** [comment]
```
<string name="pref_night_mode_title">Night mode times</string>
<string name="pref_night_mode_auto">Auto</string>
<string name="pref_night_mode_custom">Custom</string>
<string name="button_allow">Allow</string>
<string name="button_deny">Deny</string>
<string name="permission_location_rationale">QuickLyric needs your location to determine when nighttime occurs in your region.</string>
```

How do you get the translations, I always end up paying minimum price for very few words like this.

**Comment by geecko86** [comment]
We use crowdin, which is free for OSS projets.

**Comment by geecko86** [comment]
I'm not seeing your commit. Did you continue to work on the night theme like we said?

**Comment by svenoaks** [comment]
No, things happened. I'll work on it today.

**Comment by geecko86** [comment]
That's ok. Just please make sure it's done before tomorrow. We need to stay on schedule. 🙂
